engine/src/main/java/org/terasology/rendering/logic/RegionOutlineRenderer.java

```diff
@@ -0,0 +1,219 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.logic;
+
+import com.google.api.client.repackaged.com.google.common.base.Strings;
+import com.google.api.client.util.Maps;
+import com.google.common.base.Preconditions;
+import org.lwjgl.BufferUtils;
+import org.lwjgl.opengl.GL11;
+import org.terasology.assets.management.AssetManager;
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.entity.lifecycleEvents.BeforeDeactivateComponent;
+import org.terasology.entitySystem.entity.lifecycleEvents.OnActivatedComponent;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.entitySystem.systems.RenderSystem;
+import org.terasology.math.MatrixUtils;
+import org.terasology.math.Region3i;
+import org.terasology.math.geom.Matrix4f;
+import org.terasology.math.geom.Quat4f;
+import org.terasology.math.geom.Vector3f;
+import org.terasology.registry.In;
+import org.terasology.rendering.assets.material.Material;
+import org.terasology.rendering.world.WorldRenderer;
+
+import java.nio.FloatBuffer;
+import java.util.LinkedHashMap;
+
+import static org.lwjgl.opengl.GL11.GL_DEPTH_TEST;
+import static org.lwjgl.opengl.GL11.glBegin;
+import static org.lwjgl.opengl.GL11.glDisable;
+import static org.lwjgl.opengl.GL11.glEnable;
+import static org.lwjgl.opengl.GL11.glEnd;
+import static org.lwjgl.opengl.GL11.glLineWidth;
+import static org.lwjgl.opengl.GL11.glVertex3f;
+
+/**
+ * Renderes region outlines for all entities with  {@link RegionOutlineComponent}s.
+ */
+@RegisterSystem(RegisterMode.CLIENT)
+public class RegionOutlineRenderer extends BaseComponentSystem implements RenderSystem {
+
+    @In
+    private AssetManager assetManager;
+
+    @In
+    private WorldRenderer worldRenderer;
+
+    @In
+    private EntityManager entityManager;
+
+
+    private Material material;
+
+    private LinkedHashMap<EntityRef, RegionOutlineComponent> entityToRegionOutlineMap = Maps.newLinkedHashMap();
+
+    @Override
+    public void initialise() {
+        Preconditions.checkArgument(!Strings.isNullOrEmpty("engine:white"));
+        this.material = assetManager.getAsset("engine:white", Material.class).get();
+    }
+
+
+    @ReceiveEvent
+    public void onMinionActivation(OnActivatedComponent event, EntityRef entity, RegionOutlineComponent component) {
```

Comment:
`onMinionActivation` ? That's an odd use of "minion" or am I just missing some context? Typo? Auto-correct? Copy pasta? Cosmic rays? Gremlins? :-)